internal/telemetry/telemetry.go

```diff
@@ -67,14 +70,22 @@ func NewSystemUsage() (s SystemUsage) {
 	return s
 }
 
-func StartCpuUsageAverage() {
+func StartCpuUsageAverage(appWg *sync.WaitGroup, appCtx context.Context, logger logger.LoggingClient) {
 	once.Do(func() {
+		appWg.Add(1)
```

Comment:
Are we sure this goes here? Everything I read says that the .Add should be added BEFORE the goroutine to prevent raceconditions.

Reply:
Yea, I guess it should be before line 74?